CriticalMass/MapViewController.swift

```diff
@@ -183,18 +176,25 @@ class MapViewController: UIViewController {
 
     @objc func didReceiveInitialLocation(notification: Notification) {
         guard let location = notification.object as? Location else { return }
-        focusOnLocation(location: location)
+        focusOnCoordinate(CLLocationCoordinate2D(location))
 
-        // TODO: Replace test implemenation with controller based
-        guard Feature.events.isActive else { return }
         let coordinate = CLLocationCoordinate2D(
             latitude: location[keyPath: \Location.latitude],
             longitude: location[keyPath: \Location.longitude]
         )
-        nextRideHandler.getNextRide(around: coordinate) { result in
+        nextRideManager.getNextRide(around: coordinate) { result in
             switch result {
-            case let .success(rides):
-                print(rides)
+            case let .success(ride):
+                guard let nextRide = ride else {
+                    Logger.log(.debug, log: .map, "Expected nextRide to present MapInfoView")
+                    return
+                }
+                onMain { [weak self] in
+                    self?.mapInfoViewController.presentMapInfo(title: nextRide.title, style: .info)
+                    let cmAnnotation = CriticalMassAnnotation(ride: nextRide)
+                    self?.cmAnnotation = cmAnnotation
+                    self?.mapView.addAnnotation(cmAnnotation)
+                }
```

Comment:
That should be in an AnnotationController

Reply:
What exactly? The API call?

Reply:
I think now I know.
I checked the AnnotationController and now I think I know what its for 👍